File: largepower.py
mod = 1000000007;
#Method 1 :- For relatively small values. 
def power(x, y):
	result = 1;
	while(y > 0):
		if(y % 2 != 0):	result = (result * x) % mod;
		x = (x * x) % mod;
		y >>= 1;
	return result;
	
#Method 2 :- For extremely large value of the order like 10 raise to 100000.
print pow(x, y, mod);
#This takes the base x, power y and calculates result mod value.

File: largepower.c
//This function gives power of a number raised to n. Since this may result into overflow we are using mod = 1000000007
ll power(ll x, ll n){
	ll result = 1;
	while(n > 0){
		if(n % 2)	result = (result * x) % mod;
		x = (x * x) % mod, n >>= 1;
	}
	return result;
}
